storage: TestMVCCHistories failed

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11420999?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11420999?buildTab=artifacts#/) on master @ [eaf8717f2a362ad2434a0a90b598fc359b96603a](https://github.com/cockroachdb/cockroach/commits/eaf8717f2a362ad2434a0a90b598fc359b96603a):


```
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator

goroutine 4929 [running]:
testing.tRunner.func1.2({0x1ad7a60, 0x4000f09680})
	GOROOT/src/testing/testing.go:1396 +0x1c8
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x374
panic({0x1ad7a60, 0x4000f09680})
	GOROOT/src/runtime/panic.go:884 +0x20c
github.com/cockroachdb/datadriven.runDirective.func1.1()
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:334 +0x104
panic({0x1ad7a60, 0x4000f09680})
	GOROOT/src/runtime/panic.go:884 +0x20c
github.com/cockroachdb/cockroach/pkg/storage.mvccExportToWriter({0x2298008, 0x400005a068}, {0x22a4690?, 0x4000d07ca0}, {{{0x4000f16b70, 0x1, 0x8}, {0x0, 0x0, 0x0}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6653 +0xc8
github.com/cockroachdb/cockroach/pkg/storage.MVCCExportToSST({0x2298008?, 0x400005a068?}, 0xb089c0?, {0x22a4690, 0x4000d07ca0}, {{{0x4000f16b70, 0x1, 0x8}, {0x0, 0x0, ...}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6548 +0x1c0
github.com/cockroachdb/cockroach/pkg/storage_test.cmdExport(0x4000867f00)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:1490 +0x704
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1.4(0x40009fa820, 0x4000d1aad8)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:582 +0x109c
github.com/cockroachdb/datadriven.RunTest.func1({0x22b1b18?, 0x40009fa820?}, 0x40009fa820?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:118 +0x44
github.com/cockroachdb/datadriven.runDirective.func1({0x22b1b18?, 0x40009fa820?}, 0xffff995d1108?, 0x18?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:337 +0x70
github.com/cockroachdb/datadriven.runDirective({0x22b1b18, 0x40009fa820}, 0x4000d1aab0, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:342 +0x58
github.com/cockroachdb/datadriven.runDirectiveOrSubTest({0x22b1b18, 0x40009fa820}, 0x40009fa820?, {0x0, 0x0}, 0x4000011ae8?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:211 +0x84
github.com/cockroachdb/datadriven.runTestInternal({0x22b1b18, 0x40009fa820}, {0x4000d0a140, 0x131}, {0x2282000, 0x4000011ae8}, 0x698000?, 0x2?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:184 +0x88
github.com/cockroachdb/datadriven.RunTestAny({0x22b1b18, 0x40009fa820}, {0x4000d0a140, 0x131}, 0x22c6f01?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:144 +0x23c
github.com/cockroachdb/datadriven.RunTest(0x40009fa820, {0x4000d0a140, 0x131}, 0x4000b56640)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:117 +0x8c
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1(0x40009fa820, {0x4000d0a140, 0x131})
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:375 +0x52c
github.com/cockroachdb/datadriven.Walk.func1({0x22b1b18?, 0x40009fa820?}, {0x4000d0a140?, 0x131?})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:425 +0x4c
github.com/cockroachdb/datadriven.WalkAny({0x22b1b18, 0x40009fa820}, {0x4000d0a140, 0x131}, 0x40011b69c0)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:436 +0x198
github.com/cockroachdb/datadriven.WalkAny.func1({0x22b1b18, 0x40009fa820})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:449 +0x98
github.com/cockroachdb/datadriven.subTest.func1(0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/test_interface.go:27 +0x30
testing.tRunner(0x40009fa820, 0x4000b9f510)
	GOROOT/src/testing/testing.go:1446 +0x10c
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x328
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30807

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422618?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422618?buildTab=artifacts#/) on master @ [eaf8717f2a362ad2434a0a90b598fc359b96603a](https://github.com/cockroachdb/cockroach/commits/eaf8717f2a362ad2434a0a90b598fc359b96603a):

Fatal error:

```
panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator

```
Stack: 

```
goroutine 100084 [running]:
testing.tRunner.func1.2({0x1cbd680, 0xc002fcd470})
	GOROOT/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x39f
panic({0x1cbd680, 0xc002fcd470})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/datadriven.runDirective.func1.1()
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:334 +0xfc
panic({0x1cbd680, 0xc002fcd470})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/storage.mvccExportToWriter({0x247aa28, 0xc0001ac000}, {0x2486f70?, 0xc002fda3b0}, {{{0xc002fd4750, 0x1, 0x8}, {0x0, 0x0, 0x0}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6653 +0x133
github.com/cockroachdb/cockroach/pkg/storage.MVCCExportToSST({0x247aa28?, 0xc0001ac000?}, 0x100027400?, {0x2486f70, 0xc002fda3b0}, {{{0xc002fd4750, 0x1, 0x8}, {0x0, 0x0, ...}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6548 +0x2bf
github.com/cockroachdb/cockroach/pkg/storage_test.cmdExport(0xc001ad4c00)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:1490 +0x9b7
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1.4(0xc00140b860, 0xc0028e2e38)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:582 +0x15a6
github.com/cockroachdb/datadriven.RunTest.func1({0x2494478?, 0xc00140b860?}, 0x5f5fc6?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:118 +0x39
github.com/cockroachdb/datadriven.runDirective.func1({0x2494478?, 0xc00140b860?}, 0x0?, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:337 +0x89
github.com/cockroachdb/datadriven.runDirective({0x2494478, 0xc00140b860}, 0xc0028e2e10, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:342 +0x85
github.com/cockroachdb/datadriven.runDirectiveOrSubTest({0x2494478, 0xc00140b860}, 0xc00140b860?, {0x0, 0x0}, 0xc000286c08?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:211 +0x98
github.com/cockroachdb/datadriven.runTestInternal({0x2494478, 0xc00140b860}, {0xc001f0dcc0, 0x12b}, {0x2464a20, 0xc000286c08}, 0x422a0b?, 0xc0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:184 +0xa5
github.com/cockroachdb/datadriven.RunTestAny({0x2494478, 0xc00140b860}, {0xc001f0dcc0, 0x12b}, 0x101?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:144 +0x2b7
github.com/cockroachdb/datadriven.RunTest(0xc00140b860, {0xc001f0dcc0, 0x12b}, 0xc002f7e440)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:117 +0x89
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1(0xc00140b860, {0xc001f0dcc0, 0x12b})
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:375 +0x625
github.com/cockroachdb/datadriven.Walk.func1({0x2494478?, 0xc00140b860?}, {0xc001f0dcc0?, 0x12b?})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:425 +0x42
github.com/cockroachdb/datadriven.WalkAny({0x2494478, 0xc00140b860}, {0xc001f0dcc0, 0x12b}, 0xc00143be60)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:436 +0x211
github.com/cockroachdb/datadriven.WalkAny.func1({0x2494478, 0xc00140b860})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:449 +0xa7
github.com/cockroachdb/datadriven.subTest.func1(0xc0001ac000?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/test_interface.go:27 +0x27
testing.tRunner(0xc00140b860, 0xc0028ade40)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        put k=k ts=5 v=k5 localTs=4
        del_range_ts k=m end=n ts=3 localTs=2
        with t=A
          txn_begin ts=7
          put k=a v=a7
          put k=d v=d7
          put k=j v=j7
          put k=l v=l7
          put k=o v=n7
        ----
        del: "a": found key true
        del: "b": found key false
        del: "h": found key true
        >> at end:
        txn: "A" meta={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} lock=true stat=PENDING rts=7.000000000,0 wto=false gul=0,0
        rangekey: {a-b}/[1.000000000,0=/<empty>]
        rangekey: {b-c}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {c-d}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {d-f}/[5.000000000,0={localTs=4.000000000,0}/<empty> 1.000000000,0=/<empty>]
        rangekey: {f-g}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {g-h}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {h-k}/[1.000000000,0=/<empty>]
        rangekey: {m-n}/[3.000000000,0={localTs=2.000000000,0}/<empty>]
        meta: "a"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "a"/7.000000000,0 -> /BYTES/a7
        data: "a"/4.000000000,0 -> /<empty>
        data: "a"/2.000000000,0 -> /BYTES/a2
        data: "b"/4.000000000,0 -> /<empty>
        meta: "d"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "d"/7.000000000,0 -> /BYTES/d7
        data: "d"/4.000000000,0 -> /BYTES/d4
        data: "e"/3.000000000,0 -> {localTs=2.000000000,0}/BYTES/e3
        data: "f"/6.000000000,0 -> /BYTES/f6
        data: "f"/4.000000000,0 -> /BYTES/f4
        data: "f"/2.000000000,0 -> /BYTES/f2
        data: "g"/4.000000000,0 -> /BYTES/g4
        data: "g"/2.000000000,0 -> /BYTES/g2
        data: "h"/4.000000000,0 -> /<empty>
        data: "h"/3.000000000,0 -> /BYTES/h3
        meta: "j"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "j"/7.000000000,0 -> /BYTES/j7
        data: "k"/5.000000000,0 -> {localTs=4.000000000,0}/BYTES/k5
        meta: "l"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "l"/7.000000000,0 -> /BYTES/l7
        meta: "o"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "o"/7.000000000,0 -> /BYTES/n7
    datadriven.go:333: 
        panic during /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2113/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/testdata/mvcc_histories/export:81:
        export k=a end=z
    --- FAIL: TestMVCCHistories/export (0.01s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422617?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422617?buildTab=artifacts#/) on master @ [eaf8717f2a362ad2434a0a90b598fc359b96603a](https://github.com/cockroachdb/cockroach/commits/eaf8717f2a362ad2434a0a90b598fc359b96603a):

Fatal error:

```
panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator

```
Stack: 

```
goroutine 7047 [running]:
testing.tRunner.func1.2({0x1cb99e0, 0xc0014d3800})
	GOROOT/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x39f
panic({0x1cb99e0, 0xc0014d3800})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/datadriven.runDirective.func1.1()
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:334 +0xfc
panic({0x1cb99e0, 0xc0014d3800})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/storage.mvccExportToWriter({0x2475f68, 0xc000136000}, {0x24824b0?, 0xc000cf0a70}, {{{0xc0014c0528, 0x1, 0x8}, {0x0, 0x0, 0x0}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6653 +0x133
github.com/cockroachdb/cockroach/pkg/storage.MVCCExportToSST({0x2475f68?, 0xc000136000?}, 0x0?, {0x24824b0, 0xc000cf0a70}, {{{0xc0014c0528, 0x1, 0x8}, {0x0, 0x0, ...}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6548 +0x2bf
github.com/cockroachdb/cockroach/pkg/storage_test.cmdExport(0xc00374b400)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:1490 +0x9b7
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1.4(0xc0009404e0, 0xc001129df8)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:582 +0x15a6
github.com/cockroachdb/datadriven.RunTest.func1({0x248f9b8?, 0xc0009404e0?}, 0x5f5fc6?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:118 +0x39
github.com/cockroachdb/datadriven.runDirective.func1({0x248f9b8?, 0xc0009404e0?}, 0x0?, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:337 +0x89
github.com/cockroachdb/datadriven.runDirective({0x248f9b8, 0xc0009404e0}, 0xc001129dd0, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:342 +0x85
github.com/cockroachdb/datadriven.runDirectiveOrSubTest({0x248f9b8, 0xc0009404e0}, 0xc0009404e0?, {0x0, 0x0}, 0xc000014638?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:211 +0x98
github.com/cockroachdb/datadriven.runTestInternal({0x248f9b8, 0xc0009404e0}, {0xc0007ac780, 0x12b}, {0x245ffa0, 0xc000014638}, 0x422a0b?, 0x40?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:184 +0xa5
github.com/cockroachdb/datadriven.RunTestAny({0x248f9b8, 0xc0009404e0}, {0xc0007ac780, 0x12b}, 0x101?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:144 +0x2b7
github.com/cockroachdb/datadriven.RunTest(0xc0009404e0, {0xc0007ac780, 0x12b}, 0xc0006e23c0)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:117 +0x89
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1(0xc0009404e0, {0xc0007ac780, 0x12b})
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:375 +0x625
github.com/cockroachdb/datadriven.Walk.func1({0x248f9b8?, 0xc0009404e0?}, {0xc0007ac780?, 0x12b?})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:425 +0x42
github.com/cockroachdb/datadriven.WalkAny({0x248f9b8, 0xc0009404e0}, {0xc0007ac780, 0x12b}, 0xc000eb5bc0)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:436 +0x211
github.com/cockroachdb/datadriven.WalkAny.func1({0x248f9b8, 0xc0009404e0})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:449 +0xa7
github.com/cockroachdb/datadriven.subTest.func1(0xc000136000?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/test_interface.go:27 +0x27
testing.tRunner(0xc0009404e0, 0xc000391f80)
	GOROOT/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x35f
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        put k=k ts=5 v=k5 localTs=4
        del_range_ts k=m end=n ts=3 localTs=2
        with t=A
          txn_begin ts=7
          put k=a v=a7
          put k=d v=d7
          put k=j v=j7
          put k=l v=l7
          put k=o v=n7
        ----
        del: "a": found key true
        del: "b": found key false
        del: "h": found key true
        >> at end:
        txn: "A" meta={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} lock=true stat=PENDING rts=7.000000000,0 wto=false gul=0,0
        rangekey: {a-b}/[1.000000000,0=/<empty>]
        rangekey: {b-c}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {c-d}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {d-f}/[5.000000000,0={localTs=4.000000000,0}/<empty> 1.000000000,0=/<empty>]
        rangekey: {f-g}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {g-h}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {h-k}/[1.000000000,0=/<empty>]
        rangekey: {m-n}/[3.000000000,0={localTs=2.000000000,0}/<empty>]
        meta: "a"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "a"/7.000000000,0 -> /BYTES/a7
        data: "a"/4.000000000,0 -> /<empty>
        data: "a"/2.000000000,0 -> /BYTES/a2
        data: "b"/4.000000000,0 -> /<empty>
        meta: "d"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "d"/7.000000000,0 -> /BYTES/d7
        data: "d"/4.000000000,0 -> /BYTES/d4
        data: "e"/3.000000000,0 -> {localTs=2.000000000,0}/BYTES/e3
        data: "f"/6.000000000,0 -> /BYTES/f6
        data: "f"/4.000000000,0 -> /BYTES/f4
        data: "f"/2.000000000,0 -> /BYTES/f2
        data: "g"/4.000000000,0 -> /BYTES/g4
        data: "g"/2.000000000,0 -> /BYTES/g2
        data: "h"/4.000000000,0 -> /<empty>
        data: "h"/3.000000000,0 -> /BYTES/h3
        meta: "j"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "j"/7.000000000,0 -> /BYTES/j7
        data: "k"/5.000000000,0 -> {localTs=4.000000000,0}/BYTES/k5
        meta: "l"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "l"/7.000000000,0 -> /BYTES/l7
        meta: "o"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "o"/7.000000000,0 -> /BYTES/n7
    datadriven.go:333: 
        panic during /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2113/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/testdata/mvcc_histories/export:81:
        export k=a end=z
    --- FAIL: TestMVCCHistories/export (0.01s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

storage.TestMVCCHistories [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422955?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11422955?buildTab=artifacts#/) on master @ [eaf8717f2a362ad2434a0a90b598fc359b96603a](https://github.com/cockroachdb/cockroach/commits/eaf8717f2a362ad2434a0a90b598fc359b96603a):

Fatal error:

```
panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator [recovered]
	panic: interface conversion: storage_test.noopCloseReader is not storage.ReaderWithMustIterators: missing method MustEngineIterator

```
Stack: 

```
goroutine 8606 [running]:
testing.tRunner.func1.2({0x2b150c0, 0xc00089d980})
	GOROOT/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1399 +0x5f0
panic({0x2b150c0, 0xc00089d980})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/datadriven.runDirective.func1.1()
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:334 +0x165
panic({0x2b150c0, 0xc00089d980})
	GOROOT/src/runtime/panic.go:890 +0x262
github.com/cockroachdb/cockroach/pkg/storage.mvccExportToWriter({0x3866268, 0xc00013c008}, {0x38727f0?, 0xc002963320}, {{{0xc00066ddc8, 0x1, 0x8}, {0x0, 0x0, 0x0}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6653 +0x126
github.com/cockroachdb/cockroach/pkg/storage.MVCCExportToSST({0x3866268, 0xc00013c008}, 0x2f4ea38?, {0x38727f0, 0xc002963320}, {{{0xc00066ddc8, 0x1, 0x8}, {0x0, 0x0, ...}}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/storage/mvcc.go:6548 +0x347
github.com/cockroachdb/cockroach/pkg/storage_test.cmdExport(0xc000634400)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:1490 +0xdfb
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1.4(0xc0007d2340, 0xc0003061d8)
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:582 +0x1de8
github.com/cockroachdb/datadriven.RunTest.func1({0x387fcf8?, 0xc0007d2340}, 0x234b42c?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:118 +0x64
github.com/cockroachdb/datadriven.runDirective.func1({0x387fcf8, 0xc0007d2340}, 0xc0003061d8, 0xc002962010)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:337 +0x102
github.com/cockroachdb/datadriven.runDirective({0x387fcf8, 0xc0007d2340}, 0xc0003061b0, 0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:342 +0x9a
github.com/cockroachdb/datadriven.runDirectiveOrSubTest({0x387fcf8, 0xc0007d2340}, 0xc0007d2340?, {0x0, 0x0}, 0xc00058e010?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:211 +0xab
github.com/cockroachdb/datadriven.runTestInternal({0x387fcf8, 0xc0007d2340}, {0xc0000d6280, 0x12b}, {0x3850200, 0xc00058e010}, 0x494a97?, 0x4c?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:184 +0xf2
github.com/cockroachdb/datadriven.RunTestAny({0x387fcf8, 0xc0007d2340}, {0xc0000d6280, 0x12b}, 0x40?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:144 +0x3d2
github.com/cockroachdb/datadriven.RunTest(0xc0007d2340, {0xc0000d6280, 0x12b}, 0xc000476380)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:117 +0xc5
github.com/cockroachdb/cockroach/pkg/storage_test.TestMVCCHistories.func1(0xc0007d2340, {0xc0000d6280, 0x12b})
	github.com/cockroachdb/cockroach/pkg/storage_test/pkg/storage/mvcc_history_test.go:375 +0xa0a
github.com/cockroachdb/datadriven.Walk.func1({0x387fcf8?, 0xc0007d2340}, {0xc0000d6280, 0x12b})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:425 +0x6f
github.com/cockroachdb/datadriven.WalkAny({0x387fcf8, 0xc0007d2340}, {0xc0000d6280, 0x12b}, 0xc002962280)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:436 +0x2a6
github.com/cockroachdb/datadriven.WalkAny.func1({0x387fcf8, 0xc0007d2340})
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/datadriven.go:449 +0xc8
github.com/cockroachdb/datadriven.subTest.func1(0x0?)
	github.com/cockroachdb/datadriven/external/com_github_cockroachdb_datadriven/test_interface.go:27 +0x42
testing.tRunner(0xc0007d2340, 0xc000752f70)
	GOROOT/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	GOROOT/src/testing/testing.go:1493 +0x75e
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        put k=k ts=5 v=k5 localTs=4
        del_range_ts k=m end=n ts=3 localTs=2
        with t=A
          txn_begin ts=7
          put k=a v=a7
          put k=d v=d7
          put k=j v=j7
          put k=l v=l7
          put k=o v=n7
        ----
        del: "a": found key true
        del: "b": found key false
        del: "h": found key true
        >> at end:
        txn: "A" meta={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} lock=true stat=PENDING rts=7.000000000,0 wto=false gul=0,0
        rangekey: {a-b}/[1.000000000,0=/<empty>]
        rangekey: {b-c}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {c-d}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {d-f}/[5.000000000,0={localTs=4.000000000,0}/<empty> 1.000000000,0=/<empty>]
        rangekey: {f-g}/[5.000000000,0={localTs=4.000000000,0}/<empty> 3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {g-h}/[3.000000000,0=/<empty> 1.000000000,0=/<empty>]
        rangekey: {h-k}/[1.000000000,0=/<empty>]
        rangekey: {m-n}/[3.000000000,0={localTs=2.000000000,0}/<empty>]
        meta: "a"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "a"/7.000000000,0 -> /BYTES/a7
        data: "a"/4.000000000,0 -> /<empty>
        data: "a"/2.000000000,0 -> /BYTES/a2
        data: "b"/4.000000000,0 -> /<empty>
        meta: "d"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "d"/7.000000000,0 -> /BYTES/d7
        data: "d"/4.000000000,0 -> /BYTES/d4
        data: "e"/3.000000000,0 -> {localTs=2.000000000,0}/BYTES/e3
        data: "f"/6.000000000,0 -> /BYTES/f6
        data: "f"/4.000000000,0 -> /BYTES/f4
        data: "f"/2.000000000,0 -> /BYTES/f2
        data: "g"/4.000000000,0 -> /BYTES/g4
        data: "g"/2.000000000,0 -> /BYTES/g2
        data: "h"/4.000000000,0 -> /<empty>
        data: "h"/3.000000000,0 -> /BYTES/h3
        meta: "j"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "j"/7.000000000,0 -> /BYTES/j7
        data: "k"/5.000000000,0 -> {localTs=4.000000000,0}/BYTES/k5
        meta: "l"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "l"/7.000000000,0 -> /BYTES/l7
        meta: "o"/0,0 -> txn={id=00000000 key=/Min iso=Serializable pri=0.00000000 epo=0 ts=7.000000000,0 min=0,0 seq=0} ts=7.000000000,0 del=false klen=12 vlen=7 mergeTs=<nil> txnDidNotUpdateMeta=true
        data: "o"/7.000000000,0 -> /BYTES/n7
    datadriven.go:333: 
        panic during /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3320/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/storage/storage_test_/storage_test.runfiles/com_github_cockroachdb_cockroach/pkg/storage/testdata/mvcc_histories/export:81:
        export k=a end=z
    --- FAIL: TestMVCCHistories/export (0.03s)
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,race</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCHistories.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/storage/mvcc_history_test.go
+++ b/pkg/storage/mvcc_history_test.go
@@ -2683,7 +2683,7 @@ func (i *iterWithCloser) Close() {
 
 // noopCloseReader overrides Reader.Close() with a noop.
 type noopCloseReader struct {
-	storage.Reader
+	storage.ReaderWithMustIterators
 }
 
 func (noopCloseReader) Close() {}

